web/src/components/DandisetsPage.vue

```diff
@@ -103,120 +109,146 @@ export default {
       default: false,
     },
   },
-  data() {
-    return {
-      sortingOptions,
-      sortOption: Number(this.$route.query.sortOption) || 0,
-      sortDir: this.$route.query.sortDir || 1,
-      page: Number(this.$route.query.page) || 1,
-    };
-  },
-  computed: {
-    pageTitle() {
-      return (this.search) ? this.$route.query.search : this.title;
-    },
-    listingUrl() {
-      if (this.user) {
+  setup(props, ctx) {
+    // Will be replaced by `useRoute` if vue-router is upgraded to vue-router@next
+    // https://next.router.vuejs.org/api/#useroute
+    const route = ctx.root.$route;
+
+    const sortOption = ref(Number(route.query.sortOption) || 0);
+    const sortDir = ref(Number(route.query.sortDir || 1));
+    const page = ref(Number(route.query.page) || 1);
+
+    const pageTitle = computed(() => ((props.search) ? route.query.search as string : props.title));
+    const sortField = computed(() => {
+      if (toggles.DJANGO_API) {
+        return sortingOptions[sortOption.value].djangoField;
+      }
+      return sortingOptions[sortOption.value].field;
+    });
+
+    // Django dandiset listing
+
+    const djangoDandisetRequest: Ref<Paginated<Dandiset> | null> = ref(null);
+    watchEffect(async () => {
+      if (!toggles.DJANGO_API) {
+        djangoDandisetRequest.value = null;
+      } else {
+        const ordering = ((sortDir.value === -1) ? '-' : '') + sortField.value;
+        const response = await publishRest.dandisets({
+          page: page.value, page_size: DANDISETS_PER_PAGE, ordering, user: props.user ? 'me' : null,
+        });
+        djangoDandisetRequest.value = response.data;
+      }
+    });
+
+    // TODO properly type this
+    // Currently it uses the girderized representation of a dandiset version, which is untyped
+    const mostRecentDandisetVersions: Ref<any | null> = ref(null);
+    watchEffect(async () => {
+      if (djangoDandisetRequest.value === null) {
+        mostRecentDandisetVersions.value = null;
+      } else {
+        mostRecentDandisetVersions.value = await Promise.all(
+          djangoDandisetRequest.value.results.map(
+            (dandiset) => publishRest.mostRecentVersion(dandiset.identifier),
+          ),
+        );
+      }
+    });
+
+    // Girder dandiset listing
+
+    const listingUrl = computed(() => {
+      if (props.user) {
         return 'dandi/user';
       }
-      if (this.search) {
+      if (props.search) {
         return 'dandi/search';
       }
       return 'dandi';
-    },
-    pages() {
-      return Math.ceil(this.totalDandisets / DANDISETS_PER_PAGE) || 1;
-    },
-    sortField() {
-      return this.sortingOptions[this.sortOption].field;
-    },
-    queryParams() {
-      const {
-        page, sortOption, sortDir,
-      } = this;
-      return { page, sortOption, sortDir };
-    },
-    dandisets() {
+    });
+    const girderDandisetRequest: Ref<null | any> = ref(null);
```

Comment:
I believe this could be more strictly typed

```suggestion
    const girderDandisetRequest: Ref<Omit<AxiosResponse, 'config' | 'request'> | null> = ref(null);
```

You'd have to import the `AxiosResponse` type though.

Reply:
I think it's worth it to just ignore types on Girder stuff where possible because it will all be deleted shortly anyway.